code/components/gta-core-five/src/PatchPedPropsLimit.cpp

```diff
@@ -0,0 +1,107 @@
+#include <StdInc.h>
+#include <Hooking.h>
+#include "CrossBuildRuntime.h"
+
+//
+// Currently, GTA5 has a "soft limit" of 256 global (not per-DLC) ped prop indices (with a maximum index of 0xFF)
+// per anchor per model info (255 slots for `mp_f_freemode_01`, 256 for `mp_m_freemode_01`, etc.). This means that
+// with every single title update for GTA5, the number of "free" slots significantly decreases. Exceeding the
+// limit of 256 will lead to the "overflow" effect, corrupting other ped props. This happens because some specific
+// game functions expect the prop index argument to be an 8-bit integer. However, it doesn't seem to be explicitly
+// intended, as most functions expect 32-bit integers. This patch set is meant to "fix" this inconsistency and
+// make the game work fine with 32-bit global ped prop indices.
+//
+
+static HookFunction hookFunction([]()
+{
+	{
+		auto location = hook::get_pattern<char>("48 8B CD E8 ? ? ? ? 0F B6 D0 3B DA", 8);
+		hook::nop(location, 3); // nop "movzx edx, al" instruction.
+		hook::put<uint16_t>(location, 0xD08B); // put "mov edx, eax" instead.
+	}
+
+	{
+		auto location = hook::get_pattern<char>("0F B6 C8 83 C8 FF 3B D8 7D 04");
```

Comment:
Should the helmet bits in `AppearanceDataNode` being serialized with 10 bits be documented anywhere?